### PR TITLE
Fix interpreter for El Capitan, add better error output

### LINKED ISSUE #5774: Jacobian analyzer script fix for El Crapitan
It seems that ```/usr/bin/env``` in El Capitan unsets the ```DYLD_LIBRARY_PATH``` variable (this is an issue for peacock as well @permcody).

The fix is to specify the python interpreter directly (better ideas appreciated).

### PR DESCRIPTION
**Pull request by dschwen**
Closes #5774

### CODE REVIEW

**Comment by jwpeterson** [comment]
The script in 47d7ede might be handy to keep around, but after fixing a few things with TBB (we can no longer use our prebuilt 4.2 binaries with El Capitan) I think MOOSE is working just fine with the new OSX.  And we definitely do not want to change analyzejacobian.py so that it's hard-coded to miniconda...  I think this PR can be closed.

**Comment by dschwen** [comment]
Yes. Definitely.

**Comment by dschwen** [comment]
But I remember the vtk libraries and the openmp lib being a problem as
well. Hour is that fixed by changing hire tbb is linked?

**Comment by jwpeterson** [comment]
> But I remember the vtk libraries and the openmp lib being a problem as
> well. Hour is that fixed by changing hire tbb is linked?

I'll take a closer look at the VTK issue... I didn't have that module loaded when I built libmesh.

VTK issues wouldn't be fixed by fixing TBB of course, but I have to assume it's related to the same DYLD_LIBRARY_PATH issue, so there might be a bug in the way libmesh sets the rpath for that library in vtk.m4...

**Comment by permcody** [comment]
Did I catch an "Hour" or "hire" in there? Are you texting from a walkie talkie?

**Comment by dschwen** [comment]
"how". I was texting on my phone while walking.

**Comment by permcody** [comment]
I might do a SOAR observation on that behavior.

**Comment by permcody** [comment]
Did you mean "El Capitan Crunch"?

**Comment by jwpeterson** [comment]
> Are you texting from a walkie talkie?

What's your vector, victor?

**Comment by jwpeterson** [comment]
OK, I can confirm there is definitely an issue with the VTK libs.  Note the missing absolute paths on them below, which is the same issue we saw with TBB:

```
$ otool -L ../installed/bin/meshtool-opt
./installed/bin/meshtool-opt:
	/Users/moose/projects/libmesh/installed/lib/libmesh_opt.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/Users/moose/projects/libmesh/installed/lib/libnetcdf.7.dylib (compatibility version 10.0.0, current version 10.0.0)
	/usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 8.0.0)
	libvtkIOCore-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	libvtkCommonCore-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	libvtkCommonDataModel-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	libvtkFiltersCore-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	libvtkIOXML-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	libvtkImagingCore-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	libvtkIOImage-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	libvtkImagingMath-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	libvtkIOParallelXML-6.3.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
	/opt/petejw/petsc/3.6.0/lib/libpetsc.3.6.0.dylib (compatibility version 3.6.0, current version 3.6.0)
	/opt/petejw/petsc/3.6.0/lib/libparmetis.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/petejw/petsc/3.6.0/lib/libmetis.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/X11/lib/libX11.6.dylib (compatibility version 10.0.0, current version 10.0.0)
	/opt/petejw/openmpi/1.8.4/lib/libmpi_usempif08.0.dylib (compatibility version 7.0.0, current version 7.0.0)
	/opt/petejw/openmpi/1.8.4/lib/libmpi_mpifh.2.dylib (compatibility version 8.0.0, current version 8.0.0)
	/opt/moose/gcc_5.2.0/lib/libgfortran.3.dylib (compatibility version 4.0.0, current version 4.0.0)
	/opt/moose/gcc_5.2.0/lib/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/moose/gcc_5.2.0/lib/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/petejw/openmpi/1.8.4/lib/libmpi_cxx.1.dylib (compatibility version 3.0.0, current version 3.3.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
	/opt/petejw/openmpi/1.8.4/lib/libmpi.1.dylib (compatibility version 8.0.0, current version 8.0.0)
	/opt/petejw/openmpi/1.8.4/lib/libopen-rte.7.dylib (compatibility version 8.0.0, current version 8.5.0)
	/opt/petejw/openmpi/1.8.4/lib/libopen-pal.6.dylib (compatibility version 9.0.0, current version 9.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1225.1.1)
```

I think this may be an issue with the way that we build VTK.  In my notes, I have the following cmake flags:
```
cmake -DCMAKE_INSTALL_PREFIX=/opt/moose/vtk-6.2 \
      -DCMAKE_INSTALL_RPATH:STRING="/opt/moose/vtk-6.2/lib" \
      -DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON \
```
I'll ask @milljm if he adds these flags when he builds it, it's possible he doesn't.

**Comment by jwpeterson** [comment]
Yeah, his instructions are [here](http://mooseframework.org/wiki/buildingredistributable/osx_redist/osx_redist-install/vtk6/), so I think he is just missing those flags.

**Comment by dschwen** [comment]
In my executable I have
```
[10][schwd][~/Programs/INL/moose/modules/phase_field]> otool -L phase_field-opt 
phase_field-opt:
	libiomp5.dylib (compatibility version 5.0.0, current version 5.0.0)
	/Users/schwd/Programs/INL/moose/modules/phase_field/lib/libphase_field-opt.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/Users/schwd/Programs/INL/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.0.dylib (compatibility version 1.0.0, current version 1.0.0)
...
```

As I mentioned above ```libiomp5.dylib``` is a problem as well!

**Comment by jwpeterson** [comment]
> As I mentioned above libiomp5.dylib is a problem as well!

Well, we recently discovered that we can only support OpenMP in clang if we remain stuck on a specific branch from around LLVM 3.5.  Apple's system clang compiler has moved on quite a bit from this version (they call it 7.0.0 now, but that apparently doesn't correspond to a particular upstream LLVM version as far as I can tell) especially in terms of C++11 support.

Faced with this choice, we've decided to have two software stacks in the El Capitan package: one based on Apple's system clang which doesn't support OpenMP, and one based on @milljm's hand-built GCC 5.2.0 which will support OpenMP.

The libiomp5.dylib you refer to above is a remnant of our OpenMP support for clang 3.5 and won't be present in the upcoming El Capitan package.

TL;DR libiomp5.dylib won't be an issue once the new El Capitan package is released.

**Comment by friedmud** [comment]
Wait - what about OpenMP?! Why?!

OpenMP is fully integrated into Clang now... we shouldn't need to do
anything other than build a new Clang to get OpenMP support...

**Comment by friedmud** [comment]
For more information:
http://wongmichael.com/2015/08/19/clang-3-7-will-have-full-openmp-3-1-support-followed-by-openmp-4/

It's fully in Clang 3.7 (the current release).

**Comment by jwpeterson** [comment]
Derek, it doesn't work, despite what they say.  Have you tried it?  Our test code would always print that there was 1 thread, even if you explicitly call omp_set_num_threads(2).
